homeassistant/components/modbus/switch.py

```diff
@@ -209,26 +219,34 @@ def turn_off(self, **kwargs):
         if not self._verify_state:
             self._is_on = False
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self._available
+
     def update(self):
         """Update the state of the switch."""
         if not self._verify_state:
             return
 
-        value = 0
-        if self._register_type == REGISTER_TYPE_INPUT:
-            result = self._hub.read_input_registers(self._slave, self._register, 1)
-        else:
-            result = self._hub.read_holding_registers(self._slave, self._register, 1)
-
         try:
+            if self._register_type == REGISTER_TYPE_INPUT:
```

Comment:
I see now that you actually move this in the try…except. Don't do this. This code cannot raise `AttributeError` so should not be in that try…except. Only wrap code in try_except for exceptions that it can raise.